docs/01-basics/01-basic-window.md

```diff
@@ -0,0 +1,286 @@
+Find a location of your choice somewhere on your file system.
+
+Perhaps something like
+
+(Linux) `~/Projects/OpenGLGettingStarted`
+(Windows) `C:\Projects\OpenGLGettingStarted`
+
+And we shall refer to it as the project directory.
+
+Now try to reproduce the following structore of the project directory on your system
+
+```
+OpenGLGettingStarted
+├── lib
+│   └── CMakeLists.txt             # here we describe all third party dependencies
+│                                  # like glfw, glad, spdlog, imgui, ...
+├── src
+│   ├── 01-01-BasicWindow
+│   │   ├── Main.cpp               # guess what is in here
+│   │   └── CMakeLists.txt         # or here
+└── CMakeLists.txt                 # Thats our solution file
+```
+
+Next step is to fill the `CMakeLists` files, so that `CMake` knows what to do.
+
+Lets start with `lib/CMakeLists.txt`
+
+When we create our first window we have some dependencies, namely `GLFW`, `GLAD` and `spdlog`.
+`CMake` has a mechanism called [FetchContent](https://cmake.org/cmake/help/v3.28/module/FetchContent.html) which we will be using to pull those dependencies.
+
+```cmake title="lib/CMakeLists.txt"
+--8<-- "lib/CMakeLists.txt:3:5"
+```
+
+Those two line mean, make `FetchContent` available for us to use, and we would also like to be sure that our platform can handle `OpenGL` otherwise `CMake` would fail
+to prepare the project and let us know. But its unlikely that your system will not support OpenGL. We will also not be supporting anything but a somewhat modern
+Linux Distribution, like Arch or Fedora or the likes, or Windows. Operating systems from Apple will not be supported, as they lack support for the OpenGL version we are
```

Comment:
~~We will also not be supporting anything but a somewhat modern Linux Distribution, like Arch or Fedora or the likes, or Windows. Operating systems from Apple will not be supported, as they lack support for the OpenGL version we are going to target.~~
For the sake of this project we will not be targeting MacOS, it lacks the support of OpenGL version we are going to use. As well as old potato laptops from more than 10 years ago. Any current Windows and Linux OS should work.